archunit/src/main/java/com/tngtech/archunit/lang/conditions/ArchConditions.java

```diff
@@ -1421,238 +1363,8 @@ private String join(SortedSet<String> strings) {
-            boolean satisfied = endingWith.test(item);
-            String message = createMessage(item,
-                    String.format("name %s '%s'", satisfied ? "ends with" : "does not end with", suffix));
-            events.add(new SimpleConditionEvent(item, satisfied, message));
-        }
-    }
-
-    private static class DoesConditionByPredicate<T extends HasDescription & HasSourceCodeLocation>
-            extends ArchCondition<T> {
-        private final DescribedPredicate<? super T> predicate;
-
-        DoesConditionByPredicate(DescribedPredicate<? super T> predicate) {
-            super(predicate.getDescription());
-            this.predicate = predicate;
-        }
-
-        @Override
-        public void check(T item, ConditionEvents events) {
-            boolean satisfied = predicate.test(item);
-            String message = createMessage(item,
-                    (satisfied ? "does " : "does not ") + predicate.getDescription());
-            events.add(new SimpleConditionEvent(item, satisfied, message));
-        }
-    }
-
-    private static class IsConditionByPredicate<T extends HasDescription & HasSourceCodeLocation> extends ArchCondition<T> {
-        private final String eventDescription;
-        private final DescribedPredicate<T> predicate;
-
-        IsConditionByPredicate(DescribedPredicate<? super T> predicate) {
-            this(predicate.getDescription(), predicate);
-        }
-
-        IsConditionByPredicate(String eventDescription, DescribedPredicate<? super T> predicate) {
-            super(ArchPredicates.be(predicate).getDescription());
-            this.eventDescription = eventDescription;
-            this.predicate = predicate.forSubtype();
-        }
-
-        @Override
-        public void check(T member, ConditionEvents events) {
-            boolean satisfied = predicate.test(member);
-            String message = createMessage(member,
-                    (satisfied ? "is " : "is not ") + eventDescription);
-            events.add(new SimpleConditionEvent(member, satisfied, message));
-        }
-    }
-
-    private static class HaveConditionByPredicate<T extends HasDescription & HasSourceCodeLocation> extends ArchCondition<T> {
-        private final DescribedPredicate<T> rawType;
-
-        HaveConditionByPredicate(DescribedPredicate<? super T> rawType) {
-            super(ArchPredicates.have(rawType).getDescription());
-            this.rawType = rawType.forSubtype();
-        }
-
-        @Override
-        public void check(T object, ConditionEvents events) {
-            boolean satisfied = rawType.test(object);
-            String message = createMessage(object, (satisfied ? "has " : "does not have ") + rawType.getDescription());
-            events.add(new SimpleConditionEvent(object, satisfied, message));
-        }
+    private static <T extends HasDescription & HasSourceCodeLocation> ArchCondition<T> does(DescribedPredicate<? super T> predicate) {
```

Comment:
It is a little different though, because it doesn't add "does" to the message :thinking: So I though't it might be confusing since it's not consistent. That's why I kept it private... As you mention it can also not be named consistently. So for that reasons I think it's better to keep it out of the public API for now :man_shrugging: